lib/saml_idp/controller.rb

```diff
@@ -33,15 +33,21 @@ def acs_url
     end
 
     def validate_saml_request(raw_saml_request = params[:SAMLRequest])
-      decode_request(raw_saml_request)
+      decode_request(raw_saml_request, params[:Signature], params[:SigAlg], params[:RelayState])
       return true if valid_saml_request?
 
       head :forbidden if defined?(::Rails)
       false
     end
 
-    def decode_request(raw_saml_request)
-      @saml_request = Request.from_deflated_request(raw_saml_request)
+    def decode_request(raw_saml_request, signature, sig_algorithm, relay_state)
```

Comment:
There are a few notes in the README that are broken by these signature changes - can you take a quick pass at updating those?

Reply:
It might(?) just be this one

Reply:
I have checked README, there was no specific section on how `controller.rb` methods could be help. But I have added a section that explains the starting point of this gem.